### PR TITLE
2024 prediction

### PR DESCRIPTION
**Pull request by Zach3292**
## Description

Added backend math for the 2024 Crescendo game's predictions. 
Added the frontend for displaying 2024 predictions
Added all the score breakdown keys for 2024

## Motivation and Context

Allow users to see match and ranking predictions for 2024 events

## How Has This Been Tested?

All the recommended python check were done

Developped using the recommend developpement container on a m2 macbook air

Only affect the predictions and nothing else

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

**Comment by fangeugene** [comment]
I will take a detailed look this weekend, but on first pass this looks great! Thanks for the contribution.

**Comment by Zach3292** [comment]
@fangeugene Have you had a chance to take a look at it?

**Comment by phil-lopreiato** [comment]
This LGTM, but looks like it needs a rebase for merge conflicts

**Comment by Zach3292** [comment]
The merge conflicts should be fixed

**Review by andrewda — CHANGES_REQUESTED** [review]

File: src/backend/common/helpers/prediction_helper.py [hunk 1]
```diff
@@ -371,6 +371,42 @@ def calculate_before_match(self, i: int) -> TComputedMatchInfo:
...
                     count = score_breakdown[color]["totalChargeStationPoints"]
                     means[color] = count
+                elif self._stat == "note_scored":
+                    count = 0
+                    for mode in ["auto", "teleop"]:
+                        for position in ["Amp", "Speaker"]:
+                            if position == "Speaker":
+                                for type in ["NoteCount", "NoteAmplifiedCount"]:
+                                    count += score_breakdown[color][
+                                        "{}{}{}".format(mode, position, type)
```
Comment (fangeugene) [inline src/backend/common/helpers/prediction_helper.py:381]: `autoSpeakerNoteAmplifiedAcount` isn't a valid breakdown key

**Review by fangeugene — CHANGES_REQUESTED** [review]
Sorry it took me so long to get to this. This looks great, just one issue

File: src/backend/common/helpers/prediction_helper.py (hunk 1 above)
Comment (Zach3292) [reply src/backend/common/helpers/prediction_helper.py:381]: should be fixed in commit [71f1a41](https://github.com/the-blue-alliance/the-blue-alliance/pull/5746#:~:text=existant%20breakdown%20key-,71f1a41,-Merge%20state)

**Review by fangeugene — APPROVED** [review]
🥳

**Pull request merged**